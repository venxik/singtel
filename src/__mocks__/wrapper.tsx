import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
