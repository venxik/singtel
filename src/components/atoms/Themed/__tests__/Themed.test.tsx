import React from 'react';
import { Text, View } from '../Themed';
import { renderWithProviders } from '../../../../__mocks__';

describe(`Themed`, () => {
  describe('Text', () => {
    it(`match snapshot`, () => {
      const tree = renderWithProviders(<Text>Snapshot test!</Text>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('View', () => {
    it(`match snapshot`, () => {
      const tree = renderWithProviders(<View>Snapshot test!</View>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
