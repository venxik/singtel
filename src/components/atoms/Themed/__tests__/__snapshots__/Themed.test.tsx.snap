// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themed test Text match snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`Themed test View match snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</View>
`;
